opentimelineio/opentime.py

```diff
@@ -524,6 +520,53 @@ def to_frames(time_obj, fps=None):
     return int(time_obj.value_rescaled_to(fps))
 
 
+def valid_timecode_framerate(func):
+    """Decorator to make sure a valid rate is passed"""
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        valid_fps = [
+            1,
+            23.976,
+            23.98,
+            24,
+            25,
+            29.97,
+            30,
+            48,
+            59.94,
+            60
+            ]
+
+        if kwargs.get('rate') is not None:
```

Comment:
If you make this a free function, then all of these blocks become simpler and the content become much clearer I think.